### PR TITLE
web-app#307: new reset password styling

### PR DESCRIPTION
**Pull request by marvinsjsu**
https://github.com/orgs/Nonprofit-Exchange-Hub/projects/14/views/1?pane=issue&itemId=25095839

## Dev Summary

Update the password reset page based on latest mock - [password reset page](https://www.figma.com/file/K7DT5jYcLKRc9epmNrO0N7/UX---Dev-Handoff?node-id=228-5233&t=0a5133SJk6atcWfw-0)

This commit includes:
- new image to display on the password reset page
- added `id`, `name`, `onBlur`, `sublabel` props to `PasswordInput` component
- updated `SetNewPassword` component based on mocks

## Test Plan

repro steps:

1. go to http://localhost:3000/set-new-password 
2. verify that the page reflects the latest mocks
3. check if validation is applied when setting new password (error message is displayed and submit button disabled):
- if password is less than minimum # of characters
- if there's a mismatch of passwords entered

Screenshot 2023-04-24 at 11 57 18 PM

## Resources

- [stackoverflow is cool](stackoverflow.com)

### CODE REVIEW

File: client/src/assets/set-new-password.svg
Comment (esteban-gs) [inline client/src/assets/set-new-password.svg:1]: @marvinsjsu , a couple of comments.

Blocking:
- Use the transparent svg format: I sent your the transparent svg for this.
- 
Non blocking:
- See if you are be able take that and convert it to a React component through this site: https://svg2jsx.com/.  Here is a sample on how to do that: https://github.com/Nonprofit-Exchange-Hub/web-app/pull/298/files#diff-9d74a034107efe16613162b4e3a525a37cf07b4e0f2034777f731f1105efadcdR1-R13

File: client/src/components/Users/Auth/PasswordInput.tsx [hunk 1]
```diff
@@ -59,20 +75,31 @@ function PasswordInput({
...
   };
 
+  let additionalProps = {};
+
+  if (onBlur) {
+    additionalProps = {
+      onBlur,
+    };
+  }
+
```
Comment (esteban-gs) [inline client/src/components/Users/Auth/PasswordInput.tsx:85]: Just a recommendation, but non-blocking, I think we could wrap this in a function to help with readability

File: client/src/components/Users/Auth/SetNewPassword.tsx [hunk 2]
```diff
@@ -10,53 +10,135 @@ import PasswordInput from './PasswordInput';
 import { UserContext } from '../../../providers';
 import { APP_API_BASE_URL } from '../../../configs';
 
+import SetNewPasswordImg from '../../../assets/set-new-password.svg';
+
 const useStyles = makeStyles((theme: Theme) => {
-  const xPadding = 12;
+  // const xPadding = 12;
```
Comment (esteban-gs) [inline client/src/components/Users/Auth/SetNewPassword.tsx:16]: We can safely remove this, I think
```suggestion

```

File: client/src/components/Users/Auth/SetNewPassword.tsx [hunk 3]
```diff
@@ -10,53 +10,135 @@ import PasswordInput from './PasswordInput';
...
   const yPadding = 6;
   const yMargin = 8;
 
   return {
     paper: {
-      maxWidth: 821 - theme.spacing(xPadding),
+      width: '60%',
+      // maxWidth: 821 - theme.spacing(xPadding),
```
Comment (esteban-gs) [inline client/src/components/Users/Auth/SetNewPassword.tsx:23]:
```suggestion

```

File: client/src/components/Users/Auth/SetNewPassword.tsx [hunk 4]
```diff
@@ -10,53 +10,135 @@ import PasswordInput from './PasswordInput';
...
       maxHeight: 732 - theme.spacing(yPadding),
       borderRadius: 10,
       marginTop: theme.spacing(yMargin),
       marginBottom: theme.spacing(yMargin),
-      paddingTop: theme.spacing(yPadding),
+      // paddingTop: theme.spacing(yPadding),
```
Comment (esteban-gs) [inline client/src/components/Users/Auth/SetNewPassword.tsx:28]:
```suggestion

```

File: client/src/components/Users/Auth/SetNewPassword.tsx [hunk 5]
```diff
@@ -10,53 +10,135 @@ import PasswordInput from './PasswordInput';
...
       paddingBottom: theme.spacing(yPadding),
-      paddingLeft: theme.spacing(xPadding),
-      paddingRight: theme.spacing(xPadding),
+      paddingLeft: 84,
+      paddingRight: 130,
+      // paddingLeft: theme.spacing(xPadding),
+      // paddingRight: theme.spacing(xPadding),
```
Comment (esteban-gs) [inline client/src/components/Users/Auth/SetNewPassword.tsx:33]:
```suggestion
```

File: client/src/components/Users/Auth/SetNewPassword.tsx [hunk 6]
```diff
@@ -10,53 +10,135 @@ import PasswordInput from './PasswordInput';
...
+
   const [error, setError] = React.useState<string | null>(null);
+  const [confirmPasswordError, setConfirmPasswordError] = React.useState<string | null>(null);
 
   const handleChange = (evt: React.ChangeEvent<HTMLInputElement>): void => {
     setPassword(evt.target.value);
   };
 
+  const handleBlurPassword = (): void => {
+    if (password.length < PASSWORD_MIN_LENGTH) {
```
Comment (esteban-gs) [inline client/src/components/Users/Auth/SetNewPassword.tsx:96]: I think this should be 
```suggestion
    if (password.length <= PASSWORD_MIN_LENGTH) {
```

**Review by esteban-gs — APPROVED** [review]
@marvinsjsu , Thanks for putting this together. Should be good to go once the blocking comments addressed.

File: client/src/components/Users/Auth/SetNewPassword.tsx (hunk 2 above)
Comment (marvinsjsu) [reply client/src/components/Users/Auth/SetNewPassword.tsx:16]: Sounds good.  These commented out styles are removed now.

File: client/src/components/Users/Auth/PasswordInput.tsx (hunk 1 above)
Comment (marvinsjsu) [reply client/src/components/Users/Auth/PasswordInput.tsx]: Makes sense.  I've updated this so this is wrapped in a function.

**Comment by marvinsjsu** [comment]
Thanks for the review, @esteban-gs!  I've made changes to my PR and also left some responses.

File: client/src/components/Users/Auth/SetNewPassword.tsx (hunk 6 above)
Comment (esteban-gs) [reply client/src/components/Users/Auth/SetNewPassword.tsx:92]: You're right. Thanks for verifying.

File: client/src/components/Users/Auth/SetNewPassword.tsx [hunk 7]
```diff
@@ -10,53 +10,131 @@ import PasswordInput from './PasswordInput';
 import { UserContext } from '../../../providers';
 import { APP_API_BASE_URL } from '../../../configs';
 
+import SetNewPasswordImg from '../../../assets/set-new-password.svg';
+
 const useStyles = makeStyles((theme: Theme) => {
-  const xPadding = 12;
+  // const xPadding = 12;
```
Comment (esteban-gs) [inline client/src/components/Users/Auth/SetNewPassword.tsx:16]: Looks like you missed removing this comment

**Review by esteban-gs — APPROVED** [review]
Thanks for making that update. There's one comment to remove and you should be able to merge

**Pull request merged**